vms/platformvm/txs/fee/complexity.go

```diff
@@ -194,42 +187,38 @@ var (
 			wrappers.IntLen + // validators length
 			wrappers.IntLen + // subnetAuth typeID
 			wrappers.IntLen, // subnetAuthCredential typeID
-		gas.DBRead:  2, // subnet auth + manager lookup
-		gas.DBWrite: 2, // manager + weight
-		gas.Compute: 0,
+		gas.DBRead:  3, // subnet auth + transformation lookup + conversion lookup
+		gas.DBWrite: 2, // write conversion manager + total weight
 	}
 	IntrinsicRegisterSubnetValidatorTxComplexities = gas.Dimensions{
 		gas.Bandwidth: IntrinsicBaseTxComplexities[gas.Bandwidth] +
 			wrappers.LongLen + // balance
 			bls.SignatureLen + // proof of possession
 			wrappers.IntLen, // message length
-		gas.DBRead:  0, // TODO
-		gas.DBWrite: 0, // TODO
-		gas.Compute: 0, // TODO: Include PoP verification time
+		gas.DBRead:  5, // conversion owner + expiry lookup + sov lookup + subnetID/nodeID lookup + weight lookup
+		gas.DBWrite: 6, // write current staker + expiry + write weight diff + write pk diff + subnetID/nodeID lookup + weight lookup
+		gas.Compute: intrinsicBLSVerifyCompute,
 	}
 	IntrinsicSetSubnetValidatorWeightTxComplexities = gas.Dimensions{
 		gas.Bandwidth: IntrinsicBaseTxComplexities[gas.Bandwidth] +
 			wrappers.IntLen, // message length
-		gas.DBRead:  0, // TODO
-		gas.DBWrite: 0, // TODO
-		gas.Compute: 0,
+		gas.DBRead:  3, // read staker + read conversion + read weight
+		gas.DBWrite: 5, // remaining balance utxo + write weight diff + write pk diff + weights lookup + validator write
 	}
 	IntrinsicIncreaseBalanceTxComplexities = gas.Dimensions{
 		gas.Bandwidth: IntrinsicBaseTxComplexities[gas.Bandwidth] +
 			ids.IDLen + // validationID
 			wrappers.LongLen, // balance
-		gas.DBRead:  0, // TODO
-		gas.DBWrite: 0, // TODO
-		gas.Compute: 0,
+		gas.DBRead:  1, // read staker
+		gas.DBWrite: 5, // weight diff + deactivated weight diff + public key diff + delete staker + write staker
 	}
 	IntrinsicDisableSubnetValidatorTxComplexities = gas.Dimensions{
 		gas.Bandwidth: IntrinsicBaseTxComplexities[gas.Bandwidth] +
 			ids.IDLen + // validationID
 			wrappers.IntLen + // auth typeID
 			wrappers.IntLen, // authCredential typeID
-		gas.DBRead:  0, // TODO
-		gas.DBWrite: 0, // TODO
-		gas.Compute: 0,
+		gas.DBRead:  1, // read staker
+		gas.DBWrite: 6, // write remaining balance utxo + weight diff + deactivated weight diff + public key diff + delete staker + write staker
```

Comment:
could not find the weight diff and PK diff. can you point out where they are?

Reply:
https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/state/state.go#L2607